w3-aggregation.md

```diff
@@ -30,53 +30,90 @@ There are several roles in the authorization flow:
 
-### Authority
+### Aggregator
 
-_Authority_ is a [principal] that executes invoked capabilities.
+An _Aggregator_ is a type of [principal] identified by a [`did:web`] identifier.
 
-### Verifier
+An Aggregator facilitates data storage into Filecoin deals by aggregating smaller data (Filecoin Pieces) into a larger piece that can effectively be stored with a Filecoin Storage Provider.
 
-A Component of the [authority] that performs UCAN validation
+### Broker
+
+A _Broker_ is a type of [principal] identified that arranges deals for the aggregates submitted by _Storefront_.
 
 # Protocol
 
 ## Overview
 
-A Storefront is the entry point for user/application data into web3. It will act on behalf of users to move data around into different storage points. One of the key storage presences may be Filecoin Storage Providers.
+A Storefront is the entry point for user/application data into web3. It will act on behalf of users to move data around into different storage points. One of the key storage presences may be Filecoin Storage Providers. Storefront is able to ingest files of arbitrary sizes, while Storage Providers are looking for storing larger pieces, rather than small pieces. Accordingly, the aggregator is responsible for aggregating multiple smaller pieces into a bigger one that can be stored by Storage Providers.
 
 ### Authorization
 
-Broker MUST have an authorization mechanism for allowed Storefront principals (e.g. web3.storage). Either by out-of-band exchange of information or through a well defined API. For example, a broker can authorize invocations from `did:web:web3.storage` by validating the signature is from the DID. This way, it allows web3.storage to rotate keys and/or re-delegate access without having to coordinate with the broker.
+Broker MUST have an authorization mechanism for allowed Storefront principals (e.g. web3.storage). Either by out of band exchange of information or through a well defined API. For example, a broker can authorize invocations from `did:web:filecoin.web3.storage` by validating the signature is from the DID. This way, it allows web3.storage to rotate keys and/or re-delegate access without having to coordinate with the broker.
 
-### Storefront offers broker an aggregate
+### Storefront receives and verifies a piece
 
-When a Storefront has enough content to fulfill an aggregate (each broker MAY have different requirements), a Filecoin deal for an aggregate MAY be requested by an `aggregate/offer` invocation. Deal negotiations with Filecoin Storage Providers SHOULD be handled out-of-band. A broker MUST acknowledge a request by issuing a signed receipt.
+When a Storefront's user (agent) intends to store a given content into a Filecoin Storage Provider, its proof SHOULD be computed (commonly known as Filecoin Piece) by the client and added to the Storefront. The aggregator MAY decide to verify the piece before submitting it to be aggregated, depending on a out of band trust model.
```

Comment:
Do we actually need a separate `piece/verify`, I can't recall what was the rational here. As in why not do whatever verification we want as a part of `piece/add` (or whatever it may be renamed to).

Is that because we want to queue that and perform verification later ? If so maybe we should figure out general convention for that kind of thing, so we do not have to come up with creative names each time.

Reply:
One convention that comes to mind could be user submits `foo/add` using their own DID as `with`, and we then propagate using same `foo/add` but using our own `with`.  That would align with [content-claims](https://hackmd.io/@gozala/content-claims) protocol too as a general idea there is that network propagates claims by republishing them with their own DID.